microsoft/PowerToys
# [PT Run] Clear binary and json storage files on version upgrade

## Summary of the Pull Request
* There are two types of storage files in PowerToys Run - 
1. Binary Storage - used by the cache
2. Json Storage - Used for the settings.json files.

* However, these files cause inconsistencies on upgrading, especially the Binary storage files, which are binary dumps of the objects (Win32, UWP, Imagecache etc). If there is any change in the format of the object, this leads to an inconsistent cache file which needs to be cleared.
* To prevent the issue where the user has to manually delete the cache files, everytime the files are loaded during the start of PT Run, the current version is checked with the previously installed powertoys version and if it is lesser than the current Version the storage files are deleted. 
* Each file has an associated version file, based on which we decide if we should delete the file or load data from it.
* This condition can be changed once we have a portable version of powerToys.


## PR Checklist
* [x] Applies to #4424


## Detailed Description of the Pull Request / Additional comments
* The way this has been implemented is by adding a `storageHelper` class which helps to determine whether the cache must be deleted or not.
* Right now cache being deleted depends on the currentversion of powerToys and the previousVersion of PowerToys. If the previousVersion of powerToys is less than the portable version, then the cache would be deleted, provided that the previous version is not the same as the current version as we do not want to delete the cache everytime we restart PowerToys Run.
* I have also added a argument to the class, so that in the future we can choose to delete only the cache files and not the json files.


## Validation Steps Performed
* Set breakpoints to check whether the storage files are deleted or not.
* Open settings, enable, launch and disable PT run, this would delete the cache files.
* However, the next time it is run, the cache files are not cleared as the previous PT Version is the same as the current version that is being run.
